### PR TITLE
[dicttoh5] better error message for non-serializable data

### PR DESCRIPTION
**Pull request by woutdenolf**
Changelog: better error message for dicttoh5 with non-serializable data

### CODE REVIEW

**Comment by sdebionne** [comment]
This will pretty be useful, thanks!

File: src/silx/io/dictdump.py [hunk 1]
```diff
@@ -334,20 +347,28 @@ def _iter_treedict(attributes=False):
...
-                else:
-                    h5f.create_dataset(h5name,
-                                       data=data,
-                                       **create_dataset_args)
+                try:
+                    if data.shape == () or create_dataset_args is None:
+                        h5f.create_dataset(h5name, data=data)
+                    else:
+                        h5f.create_dataset(h5name, data=data, **create_dataset_args)
+                except Exception as e:
```
Comment (t20100) [inline src/silx/io/dictdump.py:364]: No way to have a stricter selection of exceptions?

**Review by t20100 — APPROVED** [review]
LGTM

File: src/silx/io/dictdump.py (hunk 1 above)
Comment (woutdenolf) [reply src/silx/io/dictdump.py:364]: I reduced it to `TypeError` only. Ok?

Comment (t20100) [reply src/silx/io/dictdump.py:364]: Yes :+1:

**Pull request merged**